lib/bootstrap.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# shellcheck disable=SC2128
+bp_dir=$(cd "$(dirname "$BASH_SOURCE")"; cd ..; pwd)
+
+# shellcheck source=/dev/null
+source "$bp_dir/lib/utils/log.sh"
+
+install_go() {
+  local go_dir="${1:?}"
+  local go_tgz
+
+  go_tgz="$(mktemp /tmp/go.tgz.XXXXXX)"
+
+  curl --retry 3 -sf -o "$go_tgz" -L https://dl.google.com/go/go1.12.9.linux-amd64.tar.gz
+  tar -C "$go_dir" -xzf "$go_tgz"
+}
+
+build_cmd() {
+  local cmd=$1
+  local layer_dir=$2
+
+  go get -d "./cmd/${cmd}/..."
+  go build -o "$layer_dir/bin/$cmd" "./cmd/${cmd}/..."
+  chmod +x "$layer_dir/bin/$cmd"
+}
+
+create_bootstrap_layer() {
+  mkdir -p "${layer_dir}"
+
+  echo "cache = true" > "${layer_dir}.toml"
+
+  {
+    echo "build = true"
+    echo "launch = false"
+  } >> "${layer_dir}.toml"
```

Comment:
Nitpick: I'm not really sure why the layer TOML is created in this way and why cache is special. If there is no specific reason behind this, it could be replaced with a Heredoc.

Reply:
I did that because I thought shellcheck prefers this format... but it actually is fine with 2 `>>`, so I'm going to change it to that

Reply:
(it reads a little nicer)